src/rgw/rgw_common.cc

```diff
@@ -70,6 +84,169 @@ void rgw_perf_stop(CephContext *cct)
+  cct->_conf->get_val_from_conf_file(sections, "percentiles", percentiles_conf_val, false);
+  cct->_conf->get_val_from_conf_file(sections, "object_sizes", object_sizes_conf_val, false);
+  cct->_conf->get_val_from_conf_file(sections, "start", p_start, false);
+  cct->_conf->get_val_from_conf_file(sections, "end", p_end, false);
+  cct->_conf->get_val_from_conf_file(sections, "inc", p_inc, false);
+  cct->_conf->get_val_from_conf_file(sections, "update", p_update, false);
+  cct->_conf->get_val_from_conf_file(sections, "reset", p_reset, false);
+
+  stringstream ss;
+  string item;
+
+  if(percentiles_conf_val.size() > 0) {
+    ss.str(percentiles_conf_val);
+    while (getline(ss, item, ',')) {
+      percentiles.push_back(std::atof(item.c_str()));
+    }
+  }
+  if(object_sizes_conf_val.size() > 0) {
+    ss.clear();
+    ss.str("");
+    item = "";
+    ss.str(object_sizes_conf_val);
+    while (getline(ss, item, ',')) {
+      object_sizes.push_back(std::atoi(item.c_str()));
+    }
+  }
+
+  if(percentiles.empty()) {
+    percentiles.push_back(0.99);
+  }
+  if(object_sizes.empty()) {
+    object_sizes.push_back(1);
+  }
+  if(p_start.size() > 0) {
+    p_starti = atoi(p_start.c_str());
+  } else {
+    p_starti = 0;
+  }
+  if(p_end.size() > 0) {
+    p_endi = atoi(p_end.c_str());
+  } else {
+    p_endi = 1000;
+  }
+  if(p_inc.size() > 0) {
+    p_inci = atoi(p_inc.c_str());
+  } else {
+    p_inci = 2;
+  }
+  if(p_update.size() > 0) {
+    p_updatei = atoi(p_update.c_str());
+  } else {
+    p_updatei = 60;
+  }
+  if(p_reset.size() > 0) {
+    p_reseti = atoi(p_reset.c_str());
+  } else {
+    p_reseti = 60 * 60;
+  }
+
+  percentile_put_first = percentile_get_first + (percentiles.size() * object_sizes.size());
+  int percentile_last = percentile_get_first + (2 * (percentiles.size() * object_sizes.size())) + 1;
+  PerfCountersBuilder plb(cct, "percentiles", percentile_get_first, percentile_last);
+
+  get_lat_centile = new centile::CentileCollection(p_starti, p_endi, p_inci, object_sizes);
+  put_lat_centile = new centile::CentileCollection(p_starti, p_endi, p_inci, object_sizes);
+
+  for(vector<unsigned int>::iterator object_size_it = object_sizes.begin(); object_size_it != object_sizes.end(); object_size_it++) {
+    for(vector<double>::iterator percentile_it = percentiles.begin(); percentile_it != percentiles.end(); percentile_it++) {
+      std::ostringstream ostr;
+      ostr << *object_size_it;
+      std::string object_size = ostr.str();
+      ostr.clear();
+      ostr.str("");
+      ostr << *percentile_it;
+      std::string percentile = ostr.str();
+      string str = string("Get_ObjectSize_") + object_size + "_Percentile_" + percentile;
+      char * cstrget = new char [str.length()+1];
+      strcpy (cstrget, str.c_str());
+      plb.add_u64_counter(percentile_get_first + index, cstrget);
+
+      str = string("Put_ObjectSize_") + object_size + "_Percentile_" + percentile;
```

Comment:
make this all lower case, just like the most of the perf stats, also percentile is usually 1 to 100 rather than 0.01 to 1.0